queries/tact/textobjects.scm

```diff
@@ -0,0 +1,357 @@
+  ("@interface"
+    "("
+    (string) @attribute.inner
+    ")") @attribute.outer)
+
+(trait_attributes
+  ("@interface"
+    "("
+    (string) @attribute.inner
+    ")") @attribute.outer)
+
+(trait_attributes
+  ("@interface"
+    "("
+    (string) @attribute.inner
+    ")") @attribute.outer)
+
+; loop.inner & outer
+; ------------------
+(while_statement) @loop.outer
+
+(while_statement
+  body: (block_statement
+    .
+    "{"
+    .
+    (_) @_start
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "loop.inner" @_start @_end)))
+
+(repeat_statement) @loop.outer
+
+(repeat_statement
+  body: (block_statement
+    .
+    "{"
+    .
+    (_) @_start
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "loop.inner" @_start @_end)))
+
+(do_until_statement) @loop.outer
+
+(do_until_statement
+  body: (block_statement
+    .
+    "{"
+    .
+    (_) @_start
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "loop.inner" @_start @_end)))
+
+(foreach_statement) @loop.outer
+
+(foreach_statement
+  body: (block_statement
+    .
+    "{"
+    .
+    (_) @_start
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "loop.inner" @_start @_end)))
+
+; conditional.inner & outer
+; -------------------------
+(if_statement) @conditional.outer
+
+(if_statement
+  consequence: (block_statement
+    .
+    "{"
+    .
+    (_) @_start
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "conditional.inner" @_start @_end)))
+
+(if_statement
+  alternative: (else_clause
+    (block_statement
+      .
+      "{"
+      .
+      (_) @_start
+      (_)? @_end
+      .
+      "}"
+      (#make-range! "conditional.inner" @_start @_end))))
+
+; block.inner & outer
+; -------------------
+(_
+  (block_statement) @block.inner) @block.outer
+
+; call.inner & outer
+; ------------------
+(method_call_expression) @call.outer
+
+(method_call_expression
+  arguments: (argument_list
+    .
+    "("
+    .
+    (_) @_start
+    (_)? @_end
+    .
+    ")"
+    (#make-range! "call.inner" @_start @_end)))
+
+(static_call_expression) @call.outer
+
+(static_call_expression
+  arguments: (argument_list
+    .
+    "("
+    .
+    (_) @_start
+    (_)? @_end
+    .
+    ")"
+    (#make-range! "call.inner" @_start @_end)))
+
+(instance_expression) @call.outer
+
+(instance_expression
+  arguments: (instance_argument_list
+    .
+    "{"
+    .
+    (_) @_start
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "call.inner" @_start @_end)))
+
+((initOf
+  name: (identifier) @_name
+  arguments: (argument_list
+    .
+    "("
+    .
+    (_) @_start
+    (_)? @_end
+    .
+    ")") @_args)
+  (#make-range! "call.outer" @_name @_args)
+  (#make-range! "call.inner" @_start @_end))
+
+; return.inner & outer
+; --------------------
+(return_statement
+  (_) @return.inner) @return.outer
+
+; number.inner
+; ------------
+(integer) @number.inner
+
+; parameter.inner & outer
+; -----------------------
+(parameter_list
+  ((_) @parameter.inner
+    .
+    ","? @parameter.outer) @parameter.outer)
```

Comment:
Sorry one last comment and I think it should be good to merge: I think here you will need to use `#make-range!` (you can see some rust examples for this maybe?)- right now (this will also be fixed after 1.0 release :sweat_smile:) the `@parameter.outer` query will only capture the comma itself. I think you will need to combine the ranges for it to work properly

Also not a huge deal but I think these queries don't apply to something like the `(receive_function)` node

Reply:
Yup, fixed those queries